File: csharp/src/CommandPattern/Compass.cs
using System.Collections.Generic;

namespace CommandPattern
{
    public class Compass
    {
        private static IList<string> _directions = new List<string>{"N", "E", "S", "W"};

        public static string GetRightTurnDirection(string direction)
        {
            if (direction == _directions[3]) return _directions[0];

            return _directions[_directions.IndexOf(direction) + 1];
        }

        public static string GetLeftTurnDirection(string direction)
        {
            if (direction == _directions[0]) return _directions[3];

            return _directions[_directions.IndexOf(direction) - 1];
        }

        public static string GetOppositeDirection(string direction)
        {
            var directionToTheRight = GetRightTurnDirection(direction);
            return GetRightTurnDirection(directionToTheRight);
        }
    }
}

File: ruby/lib/compass.rb
class Compass
  @directions = [:N, :E, :S, :W]

  def self.get_right_turn_direction(direction)
    return @directions.first if direction == @directions.last
    @directions[@directions.index(direction) + 1]
  end

  def self.get_left_turn_direction(direction)
    return @directions.last if direction == @directions.first
    @directions[@directions.index(direction) - 1]
  end

  def self.get_opposite_direction(direction)
    direction_to_the_right = get_right_turn_direction direction
    get_right_turn_direction direction_to_the_right
  end
end